pandas/core/indexes/timedeltas.py

```diff
@@ -59,23 +59,25 @@ def _td_index_cmp(opname, cls, nat_result=False):
     """
 
     def wrapper(self, other):
-        msg = "cannot compare a TimedeltaIndex with type {0}"
-        func = getattr(super(TimedeltaIndex, self), opname)
+        msg = "cannot compare a {cls} with type {typ}"
```

Comment:
same here? `self` is always a TimedeltaIndex

Reply:
can you add some tests which lock down these error messages? IOW the types that are *supposed* to appear